libraries/plugins/common_api/config_api.cpp

```diff
@@ -0,0 +1,167 @@
+    {
+        clean_config._max_blocks_history_depth
+            = options.at(get_option_name(STR(max_blocks_history_depth))).as<uint32_t>();
+    }
+    else if (options.count(get_option_name(STR(max_blocks_history_depth), true)))
+    {
+        clean_config._max_blocks_history_depth
+            = options.at(get_option_name(STR(max_blocks_history_depth), true)).as<uint32_t>();
+    }
+
+    if (options.count(get_option_name(STR(max_budgets_list_size))))
+    {
+        clean_config._max_budgets_list_size = options.at(get_option_name(STR(max_budgets_list_size))).as<uint32_t>();
+    }
+    else if (options.count(get_option_name(STR(max_budgets_list_size), true)))
+    {
+        clean_config._max_budgets_list_size
+            = options.at(get_option_name(STR(max_budgets_list_size), true)).as<uint32_t>();
+    }
+
+    if (options.count(get_option_name(STR(max_discussions_list_size))))
+    {
+        clean_config._max_discussions_list_size
+            = options.at(get_option_name(STR(max_discussions_list_size))).as<uint32_t>();
+    }
+    else if (options.count(get_option_name(STR(max_discussions_list_size), true)))
+    {
+        clean_config._max_discussions_list_size
+            = options.at(get_option_name(STR(max_discussions_list_size), true)).as<uint32_t>();
+    }
+
+    if (options.count(get_option_name(STR(lookup_limit))))
+    {
+        clean_config._lookup_limit = options.at(get_option_name(STR(lookup_limit))).as<uint32_t>();
+    }
+    else if (options.count(get_option_name(STR(lookup_limit), true)))
+    {
+        clean_config._lookup_limit = options.at(get_option_name(STR(lookup_limit), true)).as<uint32_t>();
+    }
+
+    if (options.count(get_option_name(STR(tags_to_analize_count))))
+    {
+        clean_config._tags_to_analize_count = options.at(get_option_name(STR(tags_to_analize_count))).as<uint32_t>();
+    }
+    else if (options.count(get_option_name(STR(tags_to_analize_count), true)))
+    {
+        clean_config._tags_to_analize_count
+            = options.at(get_option_name(STR(tags_to_analize_count), true)).as<uint32_t>();
+    }
+
+    // recreate to reset constants
+    config_api::_instances_by_api[api_name()].reset(new config_api(clean_config));
+}
+
+std::string config_api::get_option_name(const char* field, bool default_api)
+{
+    static const auto regex = std::regex("_");
+    std::string result;
+    result = (default_api) ? defailt_api_name : api_name();
+    result += "-";
+    result += field;
+    return std::regex_replace(result, regex, "-");
+}
+
+std::string config_api::get_option_description(const char* field, bool default_api)
+{
+    static const auto regex = std::regex("_");
+    std::string result = std::regex_replace(field, regex, " ");
+    result += " option for ";
+    result += (default_api) ? defailt_api_name : api_name();
+    return result;
+}
+
+config_api& get_api_config(std::string api_name)
+{
+    if (api_name.empty())
+        api_name = config_api::defailt_api_name;
```

Comment:
I would prefer not to modify function input parameters. It is better to create new local variable